### PR TITLE
Makefiles clean fix

### PR DESCRIPTION
**Pull request by barockobamo**
Remove Makefiles wilcard to fix clean sections because of Debian dash shell limitations.

### CODE REVIEW

File: jsref/Makefile [hunk 1]
```diff
@@ -217,8 +216,9 @@ Translate_syntax.cmo: Translate_syntax.ml Translate_syntax.mli Translate_syntax.
 	ocamlc -c $(REPLACE_STDLIB) $(PPX_FLAG) $(OCAMLCFLAGS) $<
 
 clean_mljsref:
-	sh -c "rm -f $(ML_FRONTEND_SRC)/*.mli"
-	sh -c "rm -f {,$(ML_FRONTEND_SRC)/}*.{cmo,byte}"
```
Comment (IgnoredAmbience) [inline jsref/Makefile]: I think the original clean command here was broken:  `$(ML_FRONTEND_SRC)` isn't defined anywhere in the Makefile (I think it was left in from a previous version).

{,$(ML_...)} also expanded out the empty string to give `rm -f *.cmo` etc.

The `.mli` line can be removed, and the other two altered to just remove the files in the current directory.

Thanks for fixing this.

File: jsref/Makefile [hunk 2]
```diff
@@ -217,8 +216,9 @@ Translate_syntax.cmo: Translate_syntax.ml Translate_syntax.mli Translate_syntax.
...
+	rm -f $(ML_FRONTEND_SRC)/*.mli
```
Comment (brabalan) [inline jsref/Makefile]: You seem to be missing one case, for the cmo and byte at top level (there are two curly groups in the second line).

File: jsref/Makefile [hunk 3]
```diff
@@ -258,9 +258,8 @@ $(DOC_DIR)/%.html $(DOC_DIR)/type_%.html: %.ml %.cmi $(PPX_BIN) $(ESDOCGEN) $(DU
 # Global options
 
 clean: clean_jsjsref clean_mljsref
-	sh -c "rm -f *.cmi"
-	sh -c "rm -f .depends"
-	sh -c "rm -Rf $(DOC_DIR)"
+	rm -f *.cmi *.cmo  *.unlog.js .depends
```
Comment (brabalan) [inline jsref/Makefile]: Why do you also clean the .unlog.js?

File: tools/esdocgen/Makefile [hunk 4]
```diff
@@ -7,6 +7,6 @@ all: esdocgen.cmo esdocgen.cmxs
 	ocamlopt -I +ocamldoc -shared -o $@ $<
 
 clean:
-	rm -f *.{cm{o,i,x,xs},o} || true
+	rm -f *.cmo *.cmi *.cmx *.cmxs *.o
```
Comment (brabalan) [inline tools/esdocgen/Makefile]: The `||true` is not needed?

File: jsref/Makefile (hunk 1 above)
Comment (barockobamo) [reply jsref/Makefile]: I have seen that during the fix and i was wondering if i had to remove it, il'll remove it.

File: tools/esdocgen/Makefile (hunk 4 above)
Comment (IgnoredAmbience) [reply tools/esdocgen/Makefile]: I think it was a MacOSX not supporting the `-f` flag hack, similar to the `sh -c "rm -f ..."` hacks elsewhere. Another alternative is to prefix the command line with an `-`, which tells make to ignore failures.

Comment (IgnoredAmbience) [reply tools/esdocgen/Makefile]: (The `-f` flag causes `rm` to not return an error if none of the files to be deleted exist when attempted.)

**Pull request merged**